bsg_noc/bsg_noc_links.vh

```diff
@@ -31,4 +31,27 @@
       logic [in_data_width-1:0] data;                                     \
   } in_struct_name
 
+
+// bsg_noc_wormhole
+`define bsg_wormhole_packet_width(reserved_width, x_cord_width, y_cord_width, len_width, data_width) \
+  (reserved_width+x_cord_width+y_cord_width+len_width+data_width)
+ 
+`define declare_bsg_wormhole_packet_s(width, reserved_width, x_cord_width, y_cord_width, len_width, in_struct_name) \
```

Comment:
maybe a comment as to why they need to be different (or do they?)

Reply:
From a user perspective, I would rather pass in my payload width than calculate the packet width and pass it in. Is there a reason to do this the other way?

Reply:
(This has the extra benefit of having the width and struct macro signatures match)

Reply:
Thanks prof and Dan! Honestly they do not have to be different, I can change the data structure to pass in the payload width.

Indeed, this data structure is not intended for users... It is used by wormhole router and channel tunnel,. So it might make sense to change the current data structure name to "wormhole_header_flit_s" (which has fixed width), add a new "wormhole_packet_s" for end users.

Reply:
Sounds good! Would it make sense to use a packed union, where the flit could be either the header flit or a payload flit?

Reply:
I guess we do not need a data structure for payload flit... :)

Reply:
Yeah, but couldn’t an adapter use the payload flit struct?

Reply:
I'm afraid not, the "header flit" also contains part of payload, it is not "header only" :) Typically we only have ~10 "header bits" in a header flit, all the rest is part of payload.

Reply:
Do the payload flits have any information except data?

Reply:
payload flits are data-only

Reply:
Ah, I see. I thought there was some metadata associated with the payload as well. Makes sense then!